### PR TITLE
re-enable ItTwoDomainsLoadBalancers.testApacheLoadBalancingCustomSample

### PR DESCRIPTION
**Pull request by xiancao**
ItTwoDomainsLoadBalancers.testApacheLoadBalancingCustomSample failed recently due to the change in Apache custom sample chart. 

When we have custom Apache plugin configuration file custom_mod_wl_apache.conf, we used to specify the path of the file in the chart using volumePath. Couple of days ago it was changed to use persistentVolumeClaimName:
persistentVolumeClaimName: pv-claim-name

Updated this in test and re-enable the test.

Jenkins run result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2225/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2228/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2229/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2230/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 1]
```diff
@@ -1941,4 +1946,46 @@ private void checkIngressReady(boolean isHostRouting, String ingressHost, boolea
       assertTrue(callWebAppAndWaitTillReady(curlCmd, 60));
     }
   }
+
+  /**
+   * Create PV and PVC for Apache custom configuration file in specified namespace.
+   * @param apacheNamespace namespace in which to create PVC
+   */
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:1953]: Don't we have common utility to create a generic PV/PVC. Check with Sankar

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java [hunk 2]
```diff
@@ -169,7 +169,7 @@
 
   // Apache constants
   public static final String APACHE_IMAGE_NAME = "phx.ocir.io/weblogick8s/oracle/apache";
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java:171]: Is it not available in OCR. When customer run the tests, they many not have access to OCIR.  It is an internal repository.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:1953]: I used the common utility to create PV/PVC:
createPVPVCAndVerify(v1pv, v1pvc, labelSelector, apacheNamespace);
We need to pass the V1PersistentVolume and V1PersistentVolumeClaim objects to that method.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/TestConstants.java:171]: The apache image is not in OCR. I built the image myself. I can ask Monica again whether we have official Apache image.

**Review by anpanigr — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 3]
```diff
@@ -649,6 +649,14 @@ public static boolean createImage(WitParams params) {
   }
 
   // -------------------------   pv/pvc  ---------------------------------
+  /**
+   * Get the V1PersistentVolume object in the Kubernetes cluster with specified Persistent Volume name.
+   * @param pvname the name of the Persistent Volume
+   * @return V1PersistentVolume the Persistent Volume object with specified name in Kubernetes cluster
+   */
+  public static V1PersistentVolume getPersistentVolumes(String pvname) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java:657]:
```suggestion
  public static V1PersistentVolume getPersistentVolume(String pvname) {
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 4]
```diff
@@ -672,6 +680,16 @@ public static boolean deletePersistentVolume(String name) {
     return PersistentVolume.delete(name);
   }
 
+  /**
+   * Get V1PersistentVolumeClaim object in the namespace with the specified Persistent Volume Claim name .
+   * @param namespace namespace in which to get the Persistent Volume Claim
+   * @param pvcname the name of Persistent Volume Claim
+   * @return V1PersistentVolumeClaim the Persistent Volume Claims Object in specified namespace
+   */
+  public static V1PersistentVolumeClaim getPersistentVolumeClaims(String namespace, String pvcname) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java:689]:
```suggestion
  public static V1PersistentVolumeClaim getPersistentVolumeClaim(String namespace, String pvcname) {
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 5]
```diff
@@ -1558,6 +1558,22 @@ public static V1PersistentVolumeList listPersistentVolumes(String labels) throws
     return listPersistentVolume;
   }
 
+  /**
+   * Get the V1PersistentVolume object in the Kubernetes cluster with specified Persistent Volume name.
+   * @param pvname the name of the Persistent Volume
+   * @return V1PersistentVolume the Persistent Volume object with specified name in Kubernetes cluster
+   */
+  public static V1PersistentVolume getPersistentVolumes(String pvname) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:1566]:
```suggestion
  public static V1PersistentVolume getPersistentVolume(String pvname) {
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 6]
```diff
@@ -1574,6 +1590,23 @@ public static V1PersistentVolumeClaimList listPersistentVolumeClaims(String name
     }
   }
 
+  /**
+   * Get V1PersistentVolumeClaim object in the namespace with the specified Persistent Volume Claim name .
+   * @param namespace namespace in which to get the Persistent Volume Claim
+   * @param pvcname the name of Persistent Volume Claim
+   * @return V1PersistentVolumeClaim the Persistent Volume Claims Object in specified namespace
+   */
+  public static V1PersistentVolumeClaim getPersistentVolumeClaims(String namespace, String pvcname) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:1599]:
```suggestion
  public static V1PersistentVolumeClaim getPersistentVolumeClaim(String namespace, String pvcname) {
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 7]
```diff
@@ -718,22 +722,35 @@ public static HelmParams installAndVerifyApache(String apacheNamespace,
           .httpsNodePort(httpsNodePort);
     }
 
-    if (volumePath != null && clusterNamePortMap != null) {
+    if (pvcName != null && clusterNamePortMap != null) {
       // create a custom Apache plugin configuration file named custom_mod_wl_apache.conf
-      // and put it under the directory specified in volumePath
+      // and put it under the directory specified in pv hostPath
       // this file provides a custom Apache plugin configuration to fine tune the behavior of Apache
+      V1PersistentVolumeClaim v1pvc = getPersistentVolumeClaims(apacheNamespace, pvcName);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:729]:
```suggestion
      V1PersistentVolumeClaim v1pvc = getPersistentVolumeClaim(apacheNamespace, pvcName);
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 8]
```diff
@@ -718,22 +722,35 @@ public static HelmParams installAndVerifyApache(String apacheNamespace,
...
+      assertNotNull(v1pvc);
+      assertNotNull(v1pvc.getSpec());
+      String pvName = v1pvc.getSpec().getVolumeName();
+      logger.info("Got PV {0} from PVC {1} in namespace {2}", pvName, pvcName, apacheNamespace);
+
+      V1PersistentVolume v1pv = getPersistentVolumes(pvName);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:735]:
```suggestion
      V1PersistentVolume v1pv = getPersistentVolume(pvName);
```

**Review by vanajamukkara — APPROVED** [review]
small change in method names, otherwise looks good

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 5 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 6 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 7 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 8 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: fixed

**Pull request merged**